core/local/atom/filter_ignored.js

```diff
@@ -28,24 +31,68 @@ function loop(
   channel /*: Channel */,
   opts /*: { ignore: Ignore } */
 ) /*: Channel */ {
-  const notIgnored = buildNotIgnored(opts.ignore)
+  const isIgnored = (path /*: string */, kind /*: EventKind */) =>
+    opts.ignore.isIgnored({
+      relativePath: path,
+      isFolder: kind === 'directory'
+    })
+
+  return channel.map(events => {
+    const batch = []
+
+    for (const event of events) {
+      if (event.noIgnore) {
+        batch.push(event)
+        continue
+      }
+
+      const isPathIgnored = isIgnored(event.path, event.kind)
+
+      if (event.action === 'renamed' && event.oldPath != null) {
+        const isOldPathIgnored = isIgnored(event.oldPath, event.kind)
+
+        if (isOldPathIgnored && isPathIgnored) {
+          log.debug({ event }, 'Ignored via .cozyignore')
+        } else if (isPathIgnored) {
+          const deletedEvent = movedToIgnoredPath(event)
+          batch.push(deletedEvent)
+        } else if (isOldPathIgnored) {
+          const createdEvent = movedFromIgnoredPath(event)
+          batch.push(createdEvent)
+        } else {
+          batch.push(event)
+        }
+      } else if (isPathIgnored) {
```

Comment:
Could probably be DRYed by including each predicate + corresponding (possibly made-up) event in a separate function then using `||` and pushing resulting not ignored event if any / logging ignored event otherwise.

Reply:
I thought I was DRYing up the code actually :D. I don't really see what you're thinking about though.
Let's discuss it directly.

Reply:
No blocker :ship: